src/features/lightspeed/roleExplanation.ts

```diff
@@ -0,0 +1,266 @@
+    "files",
+    "vars",
+    "defaults",
+    "meta",
+  ];
+  return files.some((file) => roleDirectories.includes(file));
+}
+
+export async function isDocumentInRole(
+  document: vscode.TextDocument,
+): Promise<boolean> {
+  const fileNameParts = document.fileName.split("/");
+  const rolesIndex = fileNameParts.findIndex((part) => part === "roles");
+  if (rolesIndex >= 0 && rolesIndex + 1 < fileNameParts.length) {
+    const dir = await fs.readdir(
+      fileNameParts.slice(0, rolesIndex + 2).join("/"),
+    );
+
+    const containsRoleDirectories = directoryContainsSomeRoleDirectories(dir);
+    if (containsRoleDirectories) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export const roleExplanation = async (extensionUri: vscode.Uri) => {
+  if (!vscode.window.activeTextEditor) {
+    return;
+  }
+  const document = vscode.window.activeTextEditor.document;
+  const documentInRole = await isDocumentInRole(document);
+
+  if (!documentInRole) {
+    return;
+  }
+
+  const explanationId = uuidv4();
+  const currentPanel = RoleExplanationPanel.createOrShow(
+    extensionUri,
+    explanationId,
+  );
+
+  lightSpeedManager.apiInstance.feedbackRequest(
+    { roleExplanation: { explanationId: explanationId } },
+    false,
+    false,
+  );
+
+  const roleName = getRoleNameFromFilePath(document.fileName);
+  const rolePath = getRoleNamePathFromFilePath(document.fileName);
+
+  const files = await getRoleYamlFiles(rolePath);
+
+  currentPanel.setContent(
+    `<div id="icons">
+        <span class="codicon codicon-loading codicon-modifier-spin"></span>
+        &nbsp;Generating the explanation for role: ${roleName}
+      </div>`,
+  );
+
+  const lightSpeedStatusbarText =
+    await lightSpeedManager.statusBarProvider.getLightSpeedStatusBarText();
+
+  let markdown = "";
+  lightSpeedManager.statusBarProvider.statusBar.text = `$(loading~spin) ${lightSpeedStatusbarText}`;
+  try {
+    await generateRoleExplanation(
+      lightSpeedManager.apiInstance,
+      files,
+      roleName,
+      explanationId,
+    ).then(async (response: ExplanationResponseParams | IError) => {
+      console.log(response);
+      if (isError(response)) {
+        const oneClickTrialProvider = getOneClickTrialProvider();
+        if (!(await oneClickTrialProvider.showPopup(response))) {
+          const errorMessage: string = `${response.message ?? UNKNOWN_ERROR} ${response.detail ?? ""}`;
+          vscode.window.showErrorMessage(errorMessage);
+          currentPanel.setContent(
+            `<p><span class="codicon codicon-error"></span>The operation has failed:<p>${errorMessage}</p></p>`,
+          );
+        }
+      } else {
+        markdown = response.content;
+        if (markdown.length === 0) {
+          markdown = "### No explanation provided.";
+        }
+        const html_snippet = marked.parse(markdown) as string;
+        currentPanel.setContent(html_snippet, true);
+      }
+    });
+  } catch (e) {
+    console.log(e);
```

Comment:
debug print?

Reply:
Same with this - copy paste from https://github.com/ansible/vscode-ansible/blob/main/src/features/lightspeed/playbookExplanation.ts#L119.  I'll remove